storage/scanner_test.go

```diff
@@ -331,7 +332,11 @@ func TestScannerDisabled(t *testing.T) {
 	s.SetDisabled(true)
 	lastScannerCount := s.Count()
 	util.SucceedsSoon(t, func() error {
-		time.Sleep(2 * time.Millisecond)
+		// We want to make sure that a complete scanner loop has time to complete
+		// between every retry iteration, so twice the scan interval is a safe
+		// value. In addition, the internal timers used by the scanner can
+		// occasionally fire with delay, so we want to be extra patient (see #8709).
+		time.Sleep(2*scanInterval + 200*time.Millisecond)
```

Comment:
I don't understand. First, if the scanner is in just the right place (e.g. the beginning of the loop), `SetDisabled` will take effect instantly and we won't see `Count` increasing right? Second, part of the point of the loop is to make sure that the scanner did get to the point where it sees the disabled flag no?